pandas/tests/frame/methods/test_asfreq.py

```diff
@@ -257,3 +258,12 @@ def test_asfreq_frequency_M_Q_Y_A_deprecated(self, freq, freq_depr):
         with tm.assert_produces_warning(FutureWarning, match=depr_msg):
             result = df.asfreq(freq=freq_depr)
         tm.assert_frame_equal(result, expected)
+
+    def test_asfreq_unsupported_freq(self):
+        index = PeriodIndex(["2020-01-01", "2021-01-01"], freq="M")
+        df = DataFrame({"a": Series([0, 1], index=index)})
+
+        with pytest.raises(
+            TypeError, match='"2MS" is not supported as a period frequency'
+        ):
+            df.asfreq(freq="2MS")
```

Comment:
Can you parametrize over a BaseOffset and the freq given as a string?

Reply:
sure, although this PR isn't really about the offset cases